homeassistant/components/neato/config_flow.py

```diff
@@ -26,7 +27,11 @@ def logger(self) -> logging.Logger:
     async def async_step_user(self, user_input: dict | None = None) -> dict:
         """Create an entry for the flow."""
         current_entries = self._async_current_entries()
-        if current_entries and CONF_TOKEN in current_entries[0].data:
+        if (
+            self.source != SOURCE_REAUTH
+            and current_entries
+            and CONF_TOKEN in current_entries[0].data
```

Comment:
Why do we have the check for token in the existing entry data?

Reply:
Already removed with latest commit ;-)

Reply:
Oh sorry this is another part of the code :-(

Reply:
Removed